assets/src/blocks/Covers/useCovers.js

```diff
@@ -53,26 +54,38 @@ export const useCovers = ({ post_types, tags, cover_type, initialRowsLimit, post
     const path = addQueryArgs('planet4/v1/get-covers', args);
 
     try {
-      const loadedCovers = await apiFetch({ path });
+      const loadedCovers = await apiFetch( { path, signal: controller.signal } );
 
-      if (loadedCovers) {
+      if(loadedCovers) {
         setCovers(loadedCovers);
       }
-
-    } catch (e) {
-      console.log(e);
-      setError(e.message);
-    } finally {
-      setLoading(false);
+    } catch (err) {
+      setError(err.message);
     }
+
+    setLoading(false);
   };
 
   useEffect(() => {
     if (!noLoading) {
-      loadCovers();
+      setController(typeof AbortController === 'undefined' ? undefined : new AbortController());
     }
   }, [cover_type, post_types, tags, posts, layout]);
 
+  useEffect(() => {
+    if(controller) {
```

Comment:
This ensures that the fetch request is always invoked with the `AbortController` everytime it's supported by the browser. In this case it fully supported.


 
![Screenshot 2022-09-23 at 14 06 46](https://user-images.githubusercontent.com/77975803/192014805-a89b60df-5cc7-45c6-bf2d-b804aafbda10.png)